### PR TITLE
Player shooting system

### LINKED ISSUE #7: Shooting system setup
Player must be able to shoot by the mouse direction. When bullet hits something it disappears.

### PR DESCRIPTION
**Pull request by 201flaviosilva**
### Description

- Created the bullet scene (currently using the Godot logo, (Blocked by #5 ));
- Implemented the player shooting system using gun system and timer;

### Image/Video

https://github.com/FlamingoFiestaStudio/OlympusGoneWild/assets/55815265/4a074c2d-4274-4f19-a025-d8710d614571

### Related Issue

**Closes #7**

### Does this PR introduce a breaking change? ⚠️

- [X] No
- [ ] Yes

### Tests

- [X] I tested all game E2E;
- [ ] I just tested some scenes;
- [ ] Not applicable.

### Check list

Before submitting this pull request, please ensure the following are completed:

- [X] [My commits follow the project conventions](https://github.com/FlamingoFiestaStudio/OlympusGoneWild/wiki/ProjectBestPractices#commit-message-convections).
- [ ] My code follows the project's coding style and conventions.
- [ ] I have updated the project documentation (if necessary).
- [ ] My code makes sense with the design.

## Additional Notes

If you do not agree with some value, let me know.

---

Thank you for your contribution to Olympus Gone Wild! Your effort is greatly appreciated, and it helps make this project better for everyone. 😄🎮🚀

### CODE REVIEW

File: player/Gun.gd
Comment (Gabriel-Sous-a) [inline player/Gun.gd:1]: I think `Weapon.gd` would be a better name for this file

File: player/Gun.gd [hunk 1]
```diff
@@ -0,0 +1,27 @@
+extends Marker2D
+
+signal shoot
+
+@export var Bullet: PackedScene
+
+@onready var gun: Marker2D = get_node(".")
```
Comment (Gabriel-Sous-a) [inline player/Gun.gd:7]: Same thing here

File: player/Player.gd [hunk 2]
```diff
@@ -34,7 +38,9 @@ func _process(_delta):
 		
 	velocity = direction * SPEED
 
-
-func _physics_process(_delta):
+func _physics_process(_delta) -> void:
 	move_and_slide()
-	
+
+func _on_gun_shoot(Bullet: PackedScene, _position: Vector2, _direction: Vector2) -> void:
```
Comment (Gabriel-Sous-a) [inline player/Player.gd:44]: same thing here

**Review by Gabriel-Sous-a — CHANGES_REQUESTED** [review]

**Review by Gabriel-Sous-a — APPROVED** [review]

**Review by Goncalo-FP — COMMENTED** [review]
I think player should be shooting slower.

**Review by LeoHabs — APPROVED** [review]
Seems fine to me

**Pull request merged**